### DISCUSSION TITLE
OS contribution from the French administration

### DISCUSSION THREAD

**Discussion by vviers** [nocodb/nocodb#2753] [General]
Hi there 👋🏼 ,

We are [a french government agency called ANCT](https://agence-cohesion-territoires.gouv.fr/) working on several public policy programs aimed at improving life in both the rural and urban parts of France. Part of our mission is to develop a toolbox in order to help public servants as well as local collectivities make better use of their data.

In this context, we are very keen to push the use of NocoDB within our administration, and we have hired a developper as well as a UX Designer (@LouisDelbosc and Aurélie Jallut) who are dedicated to designing and developing new features (as well as bugfixes) for our users.

We obviously would like to contribute this work back to the NocoDB codebase (and we would like to avoid a hard fork at all costs). We were wondering **what was the best approach in order to make our future contributions as useful and smooth as possible while keeping a fair amount of velocity on our side** ?

Louis has already opened [an Issue](https://github.com/nocodb/nocodb/issues/2565) along with [a PR](https://github.com/nocodb/nocodb/pull/2658) in order to add Excel support for export (which is a very big need for our users who are not tech-savvy enough to even know what a CSV is). **Is this way of working convenient to you moving forward ?** Or would you prefer some other canal for discussion (like your Discord server or this space) before we implement a feature (given that we are sometimes unsure which features are useful to your codebase vs. should only be an extension for us) ?

Looking forward to helping make NocoDB an even better tool :)

The [“Données et Territoires”](https://donnees.incubateur.anct.gouv.fr/) team at ANCT (cc @yohanboniface)
`donnees@anct.gouv.fr`

**Comment by o1lab** [comment]
Hey @vviers : thank you for the notes and appreciate your help contributing back to NocoDB. And super glad it is helping in governmental planning.

@[LouisDelbosc](https://github.com/LouisDelbosc)'s PR is currently under the review from what I can see.

> Is this way of working convenient to you moving forward ? 
- if its a bug fix - please go ahead and submit us a PR
- If its a small bug - please mention that you are taking it up.
- If its a bigger feature - we would like you to discuss with one of our core team members (@o1lab, @wingkwong) . Helps to decide on desired behaviour better.

**Comment by wingkwong** [comment]
From timely assistance, please feel free to contact us in Discord.
